src/diffusers/loaders.py

```diff
@@ -1144,14 +1144,14 @@ def load_lora_weights(
         self.load_lora_into_unet(
             state_dict,
             network_alphas=network_alphas,
-            unet=self.unet,
+            unet=getattr(self, self.unet_name),
             low_cpu_mem_usage=low_cpu_mem_usage,
             _pipeline=self,
         )
         self.load_lora_into_text_encoder(
             state_dict,
             network_alphas=network_alphas,
-            text_encoder=self.text_encoder,
+            text_encoder=getattr(self, self.text_encoder_name),
```

Comment:
```suggestion
            text_encoder=getattr(self, self.text_encoder_name, self.text_encoder_name),
```

Just in case it's not defined